typify/tests/schemas/merged-schemas.rs

```diff
@@ -172,4 +182,36 @@ impl ToString for NarrowNumber {
         self.0.to_string()
     }
 }
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct TrimFat {
+    pub a: serde_json::Value,
+}
+impl From<&TrimFat> for TrimFat {
+    fn from(value: &TrimFat) -> Self {
+        value.clone()
+    }
+}
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
```

Comment:
This is kind of ... not great. It makes sense though as we have to do a bunch of processing on each variant to make them comprehensible. To address this (i.e. in the future) we can do a deeper processing of the variant schemas rather than applying the light heuristics we currently do...

... on the other hand, this construction seems needlessly baroque and there might be situations where we are happy with doing our best on weird schemas.